libsolidity/codegen/ir/IRGenerationContext.cpp

```diff
@@ -97,6 +97,16 @@ string IRGenerationContext::newYulVariable()
 	return "_" + to_string(++m_varCounter);
 }
 
+string IRGenerationContext::trySuccessConditionVariable(Expression const& _expression) const
+{
+	solAssert(
+		dynamic_cast<FunctionCallAnnotation const*>(&_expression.annotation()) &&
+		static_cast<FunctionCallAnnotation const*>(&_expression.annotation())->tryCall,
```

Comment:
Why first static and then dynamic cast? I would also be fine with just ref-dynamic casting.

Reply:
I think even just a single static_cast, because the TypeChecker already ensured that expression's type to by `FunctionCall`?

Reply:
Never make any assumption about the behaviour of another module :)

Reply:
Never make any assumption about the behaviour of another module :)